code/datums/ruins/lavaland.dm

```diff
@@ -0,0 +1,175 @@
+	description = "A race of unbreathing lizards live here, that run faster \
+		than a human can, worship a broken dead city, and are capable of \
+		reproducing by something involving tentacles? Probably best to \
+		stay clear."
+	suffix = "lavaland_surface_ash_walker1.dmm"
+	cost = 20
+	allow_duplicates = FALSE
+
+/datum/map_template/ruin/lavaland/free_golem
+	name = "Free Golem Ship"
+	id = "golem-ship"
+	description = "Lumbering humanoids, made out of precious metals, move \
+		inside this ship. They frequently leave to mine more minerals, \
+		which they somehow turn into more of them. Seem very intent on \
+		research and individual liberty, and also geology based naming?"
+	cost = 25
+	suffix = "lavaland_surface_golem_ship.dmm"
+	allow_duplicates = FALSE
+
+/datum/map_template/ruin/lavaland/animal_hospital
+	name = "Animal Hospital"
+	id = "animal-hospital"
+	description = "Rats with cancer do not live very long. And the ones that \
+		wake up from cryostasis seem to commit suicide out of boredom."
+	cost = 5
+	suffix = "lavaland_surface_animal_hospital.dmm"
+	allow_duplicates = FALSE
+
+/datum/map_template/ruin/lavaland/sin
+	cost = 10
+
+/datum/map_template/ruin/lavaland/sin/envy
+	name = "Ruin of Envy"
+	id = "envy"
+	description = "When you get what they have, then you'll finally be happy."
+	suffix = "lavaland_surface_envy.dmm"
+
+/datum/map_template/ruin/lavaland/sin/gluttony
+	name = "Ruin of Gluttony"
+	id = "gluttony"
+	description = "If you eat enough, then eating will be all that you do."
+	suffix = "lavaland_surface_gluttony.dmm"
+
+/datum/map_template/ruin/lavaland/sin/greed
+	name = "Ruin of Greed"
+	id = "greed"
+	description = "Sure you don't need magical powers, but you WANT them, and \
+		that's what's important."
+	suffix = "lavaland_surface_greed.dmm"
+
+/datum/map_template/ruin/lavaland/sin/pride
+	name = "Ruin of Pride"
+	id = "pride"
+	description = "Wormhole lifebelts are for LOSERS, who you are better than."
+	suffix = "lavaland_surface_pride.dmm"
+
+/datum/map_template/ruin/lavaland/sin/sloth
+	name = "Ruin of Sloth"
+	id = "sloth"
+	description = "..."
+	suffix = "lavaland_surface_sloth.dmm"
+	cost = 0
+	allow_duplicates = FALSE
+
+/datum/map_template/ruin/lavaland/ato
+	name = "Automated Trade Outpost"
+	id = "ato"
+	description = "A sign at the front says 'Stealing is bad.'"
+	suffix = "lavaland_surface_automated_trade_outpost.dmm"
+	cost = 5
+
+/datum/map_template/ruin/lavaland/ufo_crash
+	name = "UFO Crash"
+	id = "ufo-crash"
+	description = "Turns out that keeping your abductees unconcious is really \
+		important. Who knew?"
+	suffix = "lavaland_surface_ufo_crash.dmm"
+	cost = 10
+
+/datum/map_template/ruin/lavaland/ww_vault
+	name = "Wishgranter Vault"
+	id = "ww-vault"
+	description = "Scrawled on the large double doors is both a message and a \
+		warning: 'meat grinder requires sacri...'. You're not so sure about \
+		this anymore."
+	suffix = "lavaland_surface_ww_vault.dmm"
+	cost = 35
```

Comment:
Lavaland will feel awfully empty in a round with just this vault+the trade outpost or something